cognite_toolkit/cdf_tk/run.py

```diff
@@ -146,3 +158,222 @@ def run_transformation(ToolGlobals: CDFToolConfig, external_ids: str | list[str]
             print(f"[bold red]ERROR:[/] Could not run transformation {transformation.external_id}.")
             print(e)
     return True
+
+
+def run_local_function(
+    ToolGlobals: CDFToolConfig,
+    source_path: Path,
+    external_id: str,
+    payload: str,
+    build_env: str,
+    schedule: str | None = None,
+    rebuild_env: bool = False,
+    verbose: bool = False,
+    no_cleanup: bool = False,
+) -> bool:
+    system_config = SystemConfig.load_from_directory(source_path / COGNITE_MODULES, build_env)
+    config = BuildConfigYAML.load_from_directory(source_path, build_env)
+    print(f"[bold]Building for environment {build_env} using {source_path!s} as sources...[/bold]")
+    config.set_environment_variables()
+    build_dir = tempfile.mkdtemp(prefix="build.", suffix=".tmp", dir=Path.cwd())
+    found = False
+    for root_module, root_values in config.modules.items():
+        if not isinstance(root_values, dict):
+            continue
+        for group_module, group_values in root_values.items():
+            if not isinstance(group_values, dict):
+                continue
+            for module_name in group_values:
+                if Path(source_path / f"{root_module}/{group_module}/{module_name}/functions/{external_id}").is_dir():
+                    config.environment.selected_modules_and_packages = [f"{module_name}"]
+                    found = True
+                    break
```

Comment:
I guess this can be a method on `BuildConfigYAML` with an appropriate name. As it is now, it not very readable.

Reply:
Agreed, but it's such a special case thing to modify the environment to only load one package, so I don't feel it belongs in BuildConfigYAML?